internal/middlewares/movies.go

```diff
@@ -0,0 +1,66 @@
+package middlewares
+
+import (
+	"context"
+	"net/http"
+
+	".com/internal/db"
+	".com/internal/models"
+	".com/internal/server"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/render"
+	"github.com/google/uuid"
+)
+
+// Define a custom type for the context key
+type movieContextKeyType struct{}
```

Comment:
```suggestion
type movieContextKey struct{}
```

struct is not a type , that is the convention. only add the type if you are created a type other than struct.